src/pkg/blob/dao/dao.go

```diff
@@ -49,6 +51,9 @@ type DAO interface {
 	// UpdateBlob update blob
 	UpdateBlob(ctx context.Context, blob *models.Blob) error
 
+	// UpdateBlob update blob status
+	UpdateBlobStatus(ctx context.Context, blob *models.Blob) error
```

Comment:
Can it return (int, error) for this method so that we chan deal with 404 response more clear in head blob middleware?

Reply:
it's actually inside the method, but again to add a return value.